pkg/services/ngalert/store/alert_rule_test.go

```diff
@@ -610,18 +610,32 @@ func TestIntegrationInsertAlertRules(t *testing.T) {
 		Cfg:           cfg.UnifiedAlerting,
 	}
 
-	gen := models.RuleGen
-	rules := gen.With(
-		gen.WithOrgID(orgID),
-		gen.WithIntervalMatching(store.Cfg.BaseInterval),
-	).GenerateManyRef(5)
-
-	deref := make([]models.AlertRule, 0, len(rules))
-	for _, rule := range rules {
-		deref = append(deref, *rule)
+	gen := models.RuleGen.With(
+		models.RuleGen.WithOrgID(orgID),
+		models.RuleGen.WithIntervalMatching(store.Cfg.BaseInterval),
+	)
+
+	generateRecordingRules := func(n int) []models.AlertRule {
+		rrs := gen.GenerateMany(n)
+		for i := range rrs {
+			rrs[i].Condition = ""
+			// TODO: These fields do not apply to recording rules - for now, we just use the default values of them. This is validated at the storage level.
+			// TODO: Consider making it so recording rules allow for empty string here, or use some other sentinel value in the future.
```

Comment:
When we address these TODOs it's probably also worth just making another generator for recording rules rather than the local function.

Reply:
makes sense. i just based it off the existing generator for now since that was fastest,